tests/gql_behave/tests/memgraph_V1/features/foreach.feature

```diff
@@ -66,29 +66,13 @@ Feature: Foreach
       | 4 |
     And no side effects
 
-  Scenario: Foreach shadowing in create
-    Given an empty graph
-    And having executed
-      """
-      FOREACH (i IN [1] | FOREACH (j IN [3,4] | CREATE (i {prop: j})));
-      """
-    When executing query:
-      """
-      MATCH (n) RETURN n.prop
-      """
-    Then the result should be:
-      | n.prop |
-      | 3 |
-      | 4 |
-    And no side effects
-
```

Comment:
This test should be returned. I am not sure if this is expected, but currently it throws error: "Cannot create node 'i' with labels or properties, because it is already declared"

On the other hand, following works: FOREACH (i IN [1] | FOREACH (j IN [3,4] | CREATE ( {prop: i+j})));

Reply:
Since we have specific scopes but use them globally, the query:
`FOREACH (i IN [1] | FOREACH (j IN [3,4] | CREATE ( {prop: i+j})));` 
works because `CREATE` can see the `i` from outer foreach.
For the same reason, the query:
`FOREACH (i IN [1] | FOREACH (j IN [3,4] | CREATE (i {prop: j})));`
Doesn't work because we are not using the `i` from the outer `FOREACH`, but we are redeclaring it as a node, which is not allowed. 
Hope this clarifies the confusion

Reply:
Okay, makes sense.